catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -914,7 +921,14 @@ private Double interpretOperatorEqual(Object left, Object right) {
 		}
 		if (left instanceof String && right instanceof Double) {
 			try {
-				int compareResult = Double.valueOf((String) left).compareTo((Double) right);
+				Double tempLeft = Double.parseDouble((String) left);
+				Double tempRight = (Double) right;
+				int compareResult; // default is false
```

Comment:
Please remove comment (clean code).
We're sparely using comments. They are always getting outdated and rarely add value.